src/main/java/gov/cabinetoffice/gap/applybackend/web/GrantSchemeController.java

```diff
@@ -67,36 +69,52 @@ public ResponseEntity<GetGrantSchemeWithApplicationAndAdverts> getGrantSchemeByI
             @ApiResponse(responseCode = "404", description = "No Grant Scheme found",
                     content = @Content(mediaType = "application/json")),
     })
-    public ResponseEntity<SchemeMandatoryQuestionApplicationFormInfosDto> schemeHasInternalApplication(@PathVariable Integer grantSchemeId){
-        log.debug("Checking if scheme with id {} has internal application form and if it is published",grantSchemeId);
+    public ResponseEntity<SchemeMandatoryQuestionApplicationFormInfosDto> schemeHasInternalApplication(@PathVariable Integer grantSchemeId) {
+        log.debug("Checking if scheme with id {} has internal application form and if it is published", grantSchemeId);
 
         final GrantScheme grantScheme = grantSchemeService.getSchemeById(grantSchemeId);
         log.debug("Scheme with id {} found. Getting associated advert", grantSchemeId);
 
-        log.debug("Getting Advert associated to scheme with id {}", grantSchemeId);
-        final GrantAdvert advert = grantAdvertService.getAdvertBySchemeId(grantSchemeId.toString());
-        log.debug("Advert with id {} found for scheme with id {}", advert.getId(), grantSchemeId);
-
-        final String webpageUrl = grantAdvertService.getExternalSubmissionUrl(advert);
-
-        final boolean webPageUrlIsForInternalApplications = webpageUrl.contains(environmentProperties.getFrontEndUri());
-        log.debug("Advert is pointing to an internal application form : {}", webPageUrlIsForInternalApplications);
+        final boolean hasSchemeAnApplication = grantApplicationService.doesSchemeHaveAnApplication(grantScheme);
+        final boolean hasPublishedApplication = grantApplicationService.doesSchemeHaveAPublishedApplication(grantScheme);
+        log.debug("Application form associated to scheme with Id {} has PUBLISHED status: {}", grantScheme.getId(), hasPublishedApplication);
 
         final SchemeMandatoryQuestionApplicationFormInfosDto dto = SchemeMandatoryQuestionApplicationFormInfosDto.builder()
-                .hasInternalApplication(webPageUrlIsForInternalApplications)
-                .hasPublishedInternalApplication(false)
+                .hasAdvertPublished(false)
+                .hasInternalApplication(hasSchemeAnApplication)
+                .hasPublishedInternalApplication(hasPublishedApplication)
                 .build();
 
-        if (webPageUrlIsForInternalApplications) {
-            final boolean hasPublishedApplication = grantApplicationService.doesSchemeHaveAPublishedApplication(grantScheme);
-            log.debug("Application form associated to advert with Id {} has PUBLISHED status: {}", advert.getId(), hasPublishedApplication);
-            dto.setHasPublishedInternalApplication(hasPublishedApplication);
-        }
+        checkIfAdvertExistAndHasApplyingUrlPointingToInternalApplication(grantSchemeId, dto);
```

Comment:
as per Jamie's comment
```suggestion
        checkIfAdvertExistAndIsInternal(grantSchemeId, dto);
```